lib/pf/access_filter/dhcp.pm

```diff
@@ -12,12 +12,42 @@ pf::access_filter::dhcp
 
 use strict;
 use warnings;
-use pf::api::jsonrpcclient;
+
+use pf::violation qw (violation_view_top);
+use pf::util qw(isenabled generate_session_id);
+use pf::CHI;
+use Scalar::Util qw(reftype);
+use pf::log;
 
 use base qw(pf::access_filter);
 tie our %ConfigDhcpFilters, 'pfconfig::cached_hash', 'config::DhcpFilters';
 tie our %DhcpFilterEngineScopes, 'pfconfig::cached_hash', 'FilterEngine::DhcpScopes';
 
+
+=head1 SUBROUTINES
+
+=head2 test
+
+Test all the rules
+
+=cut
+
+sub test {
```

Comment:
Is the filter engine tested on every DHCP request because there is quite a bit of stuff happening in this subroutine that could slow down the DHCP processing

Reply:
The pfdhcp cache the reply.